client/src/app/queries/sboms.ts

```diff
@@ -35,11 +36,11 @@ export const useFetchSBOMs = (
   };
 };
 
-export const useFetchSBOMById = (id?: number | string) => {
+export const useFetchSBOMById = (id?: string) => {
   const { data, isLoading, error } = useQuery({
     queryKey: [SBOMsQueryKey, id],
-    queryFn: () =>
-      id === undefined ? Promise.resolve(undefined) : getSBOMById(id),
+    queryFn: async () =>
+      id === undefined ? undefined : (await getSbom({ path: { id } })).data,
```

Comment:
@chirino I've been doing further tests. I found a problem I would really like to be addressed:
- The Axios errors are gone.

How to reproduce:
- Open the file `client/src/app/pages/sbom-details` and add this code to print the error in the code so you can it what I mean:

```js
useEffect(() => {
  if (fetchError) {
    console.log(fetchError);
  }
}, [fetchError]);
```

- Start the server using `npm run start:dev`
- Open your browser in a non existent SBOM. E.g. open your browser at http://localhost:3000/sboms/anything
- Open your browser dev tools and you will see something like the image below:

![image](https://github.com/user-attachments/assets/b68c3165-cf8a-479c-a3f6-17f8a4c86c2b)

So what is the problem?
- Due to this change, the auto retry of ReactQuery stopped working
- The Actual error returned from the server is gone. 

This is how it looks with the original code and without the autogenerated code:

![image](https://github.com/user-attachments/assets/18e93164-3c77-470a-a1b1-afc72d484073)

- Notice that the feature of auto retry of ReactQuery works, you see 3 GET errors in the console
- The original Axios error generated by the server is still there

Keeping the errors is crucial and I would be against losing them. Is there some fix you could apply please?

Reply:
Will look into this.

Reply:
This should be fixed now.  The error was not getting thrown.  Had to enable an option to get the desired behavior.